spv/verifier.go

```diff
@@ -121,7 +121,7 @@ type PaymentVerifier interface {
 
 // MerkleProofVerifier interfaces the verification of Merkle Proofs.
 type MerkleProofVerifier interface {
-	VerifyMerkleProof(context.Context, []byte) (bool, bool, error)
+	VerifyMerkleProof(context.Context, []byte) (string, bool, bool, error)
```

Comment:
I'm wondering should we have his return a struct instead, since 4 return types is a bit mad.

```
VerifyMerkleProof(context.Context, []byte) (MerkleProofValidation, error)
```

Or something.

Reply:
agree with @Tigh-Gherr , but why are we also returning the txid?

Reply:
We're returning it so we can check it matches the transaction we're validating - one of the test cases highlights that you may well have a valid proof, but is it actually for the tx you think it is for?
```go
// check proof.
response, err := mpv.VerifyMerkleProof(context.Background(), ancestor.Proof)
if response.TxID != "" && response.TxID != ancestor.Tx.TxID() {
	return ErrTxIDMismatch
}
```

Reply:
The other option would be to pass the expected txid, and return false if it doesn't match as part of the verify func. Which do you prefer?